aiohttp_session/redis_storage.py

```diff
@@ -43,9 +43,13 @@ def save_session(self, request, response, session):
             self.save_cookie(response, key,
                              max_age=session.max_age)
         else:
-            key = str(key)
-            self.save_cookie(response, key,
-                             max_age=session.max_age)
+            if session.empty:
+                self.save_cookie(response, session._mapping,
```

Comment:
It seems to have been added by [this commit ](https://github.com/aio-libs/aiohttp-session/commit/d8b8dd619ecd801f77fcf5bdeb506aca01e3ea14).